generators/client/templates/angular/src/main/webapp/app/shared/user/_user.service.ts

```diff
@@ -17,61 +17,50 @@
  limitations under the License.
 -%>
 import { Injectable } from '@angular/core';
-import { Http, Response } from '@angular/http';
+import { HttpClient, HttpResponse } from '@angular/common/http';
 import { Observable } from 'rxjs/Observable';
 
 import { SERVER_API_URL } from '../../app.constants';
 <%_ if (authenticationType !== 'oauth2') { _%>
 import { User } from './user.model';
 <%_ } _%>
-import { ResponseWrapper } from '../model/response-wrapper.model';
 import { createRequestOption } from '../model/request-util';
 
 @Injectable()
 export class UserService {
     private resourceUrl = SERVER_API_URL + '<%- apiUaaPath %>api/users';
 
-    constructor(private http: Http) { }
+    constructor(private http: HttpClient) { }
 <%_ if (authenticationType !== 'oauth2') { _%>
 
-    create(user: User): Observable<ResponseWrapper> {
-        return this.http.post(this.resourceUrl, user)
-            .map((res: Response) => this.convertResponse(res));
+    create(user: User): Observable<HttpResponse<User>> {
+        return this.http.post<User>(this.resourceUrl, user, { observe: 'response' });
     }
 
-    update(user: User): Observable<ResponseWrapper> {
-        return this.http.put(this.resourceUrl, user)
-            .map((res: Response) => this.convertResponse(res));
+    update(user: User): Observable<HttpResponse<User>> {
+        return this.http.put<HttpResponse<User>>(this.resourceUrl, user);
     }
 
     find(login: string): Observable<User> {
-        return this.http.get(`${this.resourceUrl}/${login}`).map((res: Response) => res.json());
+        return this.http.get(`${this.resourceUrl}/${login}`);
```

Comment:
get\<User\>

Reply:
yep missed it here like i said not sure if i added types everywhere